src/vehicles/RosVehicle.lua

```diff
@@ -151,24 +161,30 @@ function RosVehicle:getLaserFrameNode()
     local spec = self.spec_rosVehicle
 
     if self.spec_drivable then
-        local current_veh_name = ros.Names.sanatize(self:getFullName())
-        spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle[current_veh_name])
 
-        local instance_veh = VehicleCamera:new(self, RosVehicle)
-        local xml_path = instance_veh.vehicle.configFileName
-        local xmlFile = loadXMLFile("vehicle", xml_path)
+        -- if there is no custom laser scanner setting for this vehicle, use the default settings to initialize an object of LaserScanner class
+        -- note: a laser scanner is always mounted in the default settings
+        if not mod_config.vehicle[spec.ros_veh_name] then
+            spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle["default_vehicle"])
+        -- if the custom laser scanner is mounted (parameter is_mounted = true), initialize an object of LaserScanner class
+        elseif mod_config.vehicle[spec.ros_veh_name].laser_scan.is_mounted then
+            spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle[spec.ros_veh_name])
+        -- if the custom laser scanner is not mounted (parameter is_mounted = false), do not initialize an object of LaserScanner class
+        else
+            return
+        end
         --  get the cameraRaycast node 2(on top of ) which is 0 index .raycastNode(0)
         --  get the cameraRaycast node 3 (in the rear) which is 1 index .raycastNode(1)
         local cameraKey = string.format("vehicle.enterable.cameras.camera(%d).%s", 0, spec.laser_scan_obj.vehicle_table.laser_scan.laser_attachments)
-        XMLUtil.checkDeprecatedXMLElements(xmlFile, xml_path, cameraKey .. "#index", "#node") -- FS17 to FS19
-        local camIndexStr = getXMLString(xmlFile, cameraKey .. "#node")
-        instance_veh.cameraNode =
+        XMLUtil.checkDeprecatedXMLElements(spec.xmlFile, spec.xml_path, cameraKey .. "#index", "#node") -- FS17 to FS19
+        local camIndexStr = getXMLString(spec.xmlFile, cameraKey .. "#node")
+        spec.instance_veh.cameraNode =
             I3DUtil.indexToObject(
-            instance_veh.vehicle.components,
+            spec.instance_veh.vehicle.components,
             camIndexStr,
-            instance_veh.vehicle.i3dMappings
+            spec.instance_veh.vehicle.i3dMappings
         )
-        if instance_veh.cameraNode == nil then
+        if spec.instance_veh.cameraNode == nil then
```

Comment:
Is this something which could done once?

Or is it possible for a vehicle to suddenly not have a `cameraNode` any more?

Reply:
Yes, it is something could be done once. I've rearranged the structure a bit.